common/src/ray_tracing.rs

```diff
@@ -1,13 +1,26 @@
-use crate::math;
+use crate::{
+    dodeca::Vertex,
+    math,
+    node::{Chunk, ChunkId, DualGraph, VoxelData},
+    world::Material,
+};
 
-/// Performs ray tracing against the z=0 plane in node coordinates. This plane is one-sided, so the ray must be going in the -z
-/// direction to detect an intersection.
+/// Performs ray tracing against the voxels in the `DualGraph`. This function is suitable for collision checking with
+/// any different collider shape. Each collider would have a different implemenation of `ChunkRayTracer`, which describes
```

Comment:
Is this a ray trace or a shape cast (aka swept collision query)? The name suggests the former, but this comment, and the presence of collision shape information other than the voxels, suggests the latter.

Reply:
This is a shape cast, assuming the collider used isn't a point collider. I am not familiar with the terms shape-casting and "swept collision query", so ray tracing was the best term I could come up with.

I've been visualizing it as ray tracing against an abstract shape where a constant-width outline is added around all the voxels.

Reply:
Let's call it "shape cast" then.